identity/service_principals.go

```diff
@@ -0,0 +1,300 @@
+}
+
+// Delete will delete the servicePrincipal given the servicePrincipal id
+func (a ServicePrincipalsAPI) Delete(servicePrincipalID string) error {
+	servicePrincipalPath := fmt.Sprintf("/preview/scim/v2/ServicePrincipals/%v", servicePrincipalID)
+	return a.C.Scim(http.MethodDelete, servicePrincipalPath, nil, nil)
+}
+
+// SetServicePrincipalAsAdmin will add the servicePrincipal to a admin group given the admin group id and servicePrincipal id
+func (a ServicePrincipalsAPI) SetServicePrincipalAsAdmin(servicePrincipalID string, adminGroupID string) error {
+	servicePrincipalPath := fmt.Sprintf("/preview/scim/v2/ServicePrincipals/%v", servicePrincipalID)
+	var addOperations ServicePrincipalPatchOperations
+	servicePrincipalPatchRequest := ServicePrincipalPatchRequest{
+		Schemas:    []URN{PatchOp},
+		Operations: []ServicePrincipalPatchOperations{},
+	}
+	addOperations = ServicePrincipalPatchOperations{
+		Op: "add",
+		Value: &GroupsValue{
+			Groups: []ValueListItem{{Value: adminGroupID}},
+		},
+	}
+	servicePrincipalPatchRequest.Operations = append(servicePrincipalPatchRequest.Operations, addOperations)
+	return a.C.Scim(http.MethodPatch, servicePrincipalPath, servicePrincipalPatchRequest, nil)
+}
+
+// VerifyServicePrincipalAsAdmin will verify the servicePrincipal belongs to the admin group given the admin group id and servicePrincipal id
+func (a ServicePrincipalsAPI) VerifyServicePrincipalAsAdmin(servicePrincipalID string, adminGroupID string) (bool, error) {
+	servicePrincipal, err := a.read(servicePrincipalID)
+	if err != nil {
+		return false, err
+	}
+	for _, group := range servicePrincipal.Groups {
+		if group.Value == adminGroupID {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// RemoveServicePrincipalAsAdmin will remove the servicePrincipal from the admin group given the admin group id and servicePrincipal id
+func (a ServicePrincipalsAPI) RemoveServicePrincipalAsAdmin(servicePrincipalID string, adminGroupID string) error {
+	servicePrincipalPath := fmt.Sprintf("/preview/scim/v2/ServicePrincipals/%v", servicePrincipalID)
+	var removeOperations ServicePrincipalPatchOperations
+	servicePrincipalPatchRequest := ServicePrincipalPatchRequest{
+		Schemas:    []URN{PatchOp},
+		Operations: []ServicePrincipalPatchOperations{},
+	}
+	path := fmt.Sprintf("groups[value eq \"%s\"]", adminGroupID)
+	removeOperations = ServicePrincipalPatchOperations{
+		Op:   "remove",
+		Path: path,
+	}
+	servicePrincipalPatchRequest.Operations = append(servicePrincipalPatchRequest.Operations, removeOperations)
+	return a.C.Scim(http.MethodPatch, servicePrincipalPath, servicePrincipalPatchRequest, nil)
+}
+
+// GetOrCreateDefaultMetaServicePrincipal ...
+func (a ServicePrincipalsAPI) GetOrCreateDefaultMetaServicePrincipal(metaServicePrincipalDisplayName string, metaServicePrincipalName string, deleteAfterCreate bool) (servicePrincipal ScimServicePrincipal, err error) {
```

Comment:
remove this one

Reply:
2c313ea0d30a96f5ec12d5aa1572c2fc02c32bd9